src/webots/nodes/WbCharger.cpp

```diff
@@ -199,7 +199,7 @@ void WbCharger::prePhysicsStep(double ms) {
     } else
       currentEnergy -= e;  // transfer
 
-    if (mDone == false && mRobot->battery().size() > 2) {  // else energy is wasted
+    if (mDone == false) {  // else energy is wasted
```

Comment:
I am not sure this is correct. We should probably modify line 190 instead.

Reply:
After discussion, this is actually correct.